x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rule_snooze_panel.tsx

```diff
@@ -0,0 +1,604 @@
+      snoozeRule={onSnoozeRule}
+      unsnoozeRule={onUnsnoozeRule}
+      interval={interval}
+      showCancel={showCancel}
+      scheduledSnoozes={scheduledSnoozes}
+      navigateToScheduler={onOpenScheduler}
+      onRemoveAllSchedules={cancelSnoozeSchedules}
+      hasTitle={hasTitle}
+    />
+  ) : (
+    <RuleSnoozeScheduler
+      isLoading={isLoading}
+      initialSchedule={initialSchedule}
+      onClose={onCloseScheduler}
+      onSaveSchedule={saveSnoozeSchedule}
+      onCancelSchedules={cancelSnoozeSchedules}
+      hasTitle={hasTitle}
+    />
+  );
+};
+
+const PREV_SNOOZE_INTERVAL_KEY = 'triggersActionsUi_previousSnoozeInterval';
+export const usePreviousSnoozeInterval: (
+  p?: string | null
+) => [string | null, (n: string) => void] = (propsInterval) => {
+  const intervalFromStorage = localStorage.getItem(PREV_SNOOZE_INTERVAL_KEY);
+  const usePropsInterval = typeof propsInterval !== 'undefined';
+  const interval = usePropsInterval ? propsInterval : intervalFromStorage;
+  const [previousSnoozeInterval, setPreviousSnoozeInterval] = useState<string | null>(interval);
+  const storeAndSetPreviousSnoozeInterval = (newInterval: string) => {
+    if (!usePropsInterval) {
+      localStorage.setItem(PREV_SNOOZE_INTERVAL_KEY, newInterval);
+    }
+    setPreviousSnoozeInterval(newInterval);
+  };
+  return [previousSnoozeInterval, storeAndSetPreviousSnoozeInterval];
+};
+
+const BaseSnoozePanel: React.FunctionComponent<BaseSnoozePanelProps> = ({
+  isLoading,
+  interval = '3d',
+  snoozeRule,
+  unsnoozeRule,
+  showCancel,
+  scheduledSnoozes,
+  navigateToScheduler,
+  onRemoveAllSchedules,
+  hasTitle,
+}) => {
+  const [intervalValue, setIntervalValue] = useState(parseInterval(interval).value);
+  const [intervalUnit, setIntervalUnit] = useState(parseInterval(interval).unit);
+
+  const [isRemoveAllModalVisible, setIsRemoveAllModalVisible] = useState(false);
+
+  const [previousSnoozeInterval, setPreviousSnoozeInterval] = usePreviousSnoozeInterval();
+
+  const onChangeValue = useCallback(
+    ({ target }) => setIntervalValue(target.value),
+    [setIntervalValue]
+  );
+  const onChangeUnit = useCallback(
+    ({ target }) => setIntervalUnit(target.value),
+    [setIntervalUnit]
+  );
+
+  const snoozeRuleAndStoreInterval = useCallback(
+    (newSnoozeEndTime: string | -1, intervalToStore: string | null) => {
+      if (intervalToStore) {
+        setPreviousSnoozeInterval(intervalToStore);
+      }
+      const newSnoozeSchedule = {
+        id: null,
+        duration: newSnoozeEndTime === -1 ? -1 : Date.parse(newSnoozeEndTime) - Date.now(),
+        rRule: { dtstart: new Date().toISOString(), count: 1, tzid: moment.tz.guess() },
+      };
+      return snoozeRule(newSnoozeSchedule);
+    },
+    [setPreviousSnoozeInterval, snoozeRule]
+  );
+
+  const applySnooze = useCallback(
+    async (value: number, unit?: SnoozeUnit) => {
+      if (value === -1) {
+        await snoozeRuleAndStoreInterval(-1, null);
```

Comment:
do we need to manage some kind of error, like having a try/catch and maybe show toaster to show errors

@mdefazio what do you think?

Reply:
I think we should show a toast when there's an error. Probably also show a toast when successful too :) 

What do you mean by manage though?  I don't think we should have a different state for this button/popover when there was an error.  In https://github.com/elastic/kibana/pull/133641 the saving animation is moved to the row (closes popover). Maybe i'm not following you though...

Reply:
manage === taking care